// api-alphabot/main.js
const zee = require('./index.js')

module.exports.Palingmurah = zee.Palingmurah
module.exports.Moddroid = zee.Moddroid
module.exports.Apkmody = zee.Apkmody
module.exports.Happymod = zee.Happymod
module.exports.Wpsearch = zee.Wpsearch
module.exports.Muihalal = zee.Muihalal
module.exports.Mcpedl = zee.Mcpedl
module.exports.Servermc = zee.Servermc
module.exports.Pstore = zee.Pstore
module.exports.Jalantikus = zee.Jalantikus
module.exports.Sfilesearch = zee.Jfilesearch
module.exports.Turnbackhoax = zee.Turnbackhoax
module.exports.Igfoto = zee.Igfoto
module.exports.Igvideo = zee.Igvideo
module.exports.Igtv = zee.Igtv
module.exports.Tribunnews = zee.Tribunnews
module.exports.Igstory = zee.Igstory
module.exports.Ttdownloader = zee.Ttdownloader
module.exports.Igdl = zee.Igdl
module.exports.Youtube = zee.Downloader
module.exports.Tiktok = zee.Downloader
module.exports.Instagram = zee.Downloader
module.exports.Facebook = zee.Downloader
module.exports.Twitter = zee.Downloader
module.exports.SoundCloud = zee.Downloader
module.exports.Imgur = zee.Downloader
module.exports.Imdb = zee.Downloader
module.exports.Anime = zee.Anime
module.exports.Manga = zee.Manga
module.exports.Character = zee.Character
module.exports.JadwalBola = zee.JadwalBola
module.exports.JadwalTv = zee.JadwalTv
module.exports.JadwalSholat = zee.JadwalSholat
module.exports.Pinterest = zee.Pinterest
module.exports.Film = zee.Film
module.exports.Wattpad = zee.Wattpad
module.exports.WattpadUser = zee.WattpadUser
module.exports.Webtoons = zee.Webtoons
module.exports.Mangatoons = zee.Mangatoons
module.exports.Drakor = zee.Drakor
module.exports.Telesticker = zee.Telesticker
module.exports.StickerSearch = zee.StickerSearch
module.exports.ListSurah = zee.ListSurah
module.exports.Surah = zee.Surah
module.exports.TafsirSurah = zee.TafsirSurah
module.exports.KompasNews = zee.KompasNews
module.exports.INews = zee.INews
